components/server/index.js

```diff
@@ -31,16 +35,20 @@ export class ServerComponent {
       componentsUsed.add("legacy");
     }
 
-    const res = new this().render(...args);
-
-    if (!res || res === nothing) {
+    const renderResult = new this().render(...args);
```

Comment:
I think it's okay to use `let` in this case, and overwrite it below:

```suggestion
    let renderResult = new this().render(...args);
```